sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidJoinQueryRel.java

```diff
@@ -328,17 +327,27 @@ private static JoinType toDruidJoinType(JoinRelType calciteJoinType)
     }
   }
 
-  private static boolean computeLeftRequiresSubquery(final DruidRel<?> left)
+  private static boolean computeLeftRequiresSubquery(final List<DruidRel<?>> leftList)
```

Comment:
What do you think about keeping this interface the same and instead changing the private members left and right to `DruidRel<?>`

Whenever left and right is set - we can do something like `this.left = getCheapestDruidRel(RelNode);`

As it's written, I wonder if it's possible for us to estimate that a subquery is not required in the computeSelfCost function, but the DruidRel used to build the native query actually uses a subquery.